utilities/user_utils.py

```diff
@@ -0,0 +1,263 @@
+import secrets
+import string
+import random
+import logging
+import tempfile
+import subprocess
+import os
+import json
+from dataclasses import dataclass
+from pyhelper_utils.shell import run_command
+from timeout_sampler import TimeoutSampler
+from utilities.infra import login_with_user_password
+
+
+LOGGER = logging.getLogger(__name__)
+SLEEP_TIME = 5
+
+
+@dataclass
+class UserTestSession:
+    """Represents a test user session with all necessary credentials and contexts."""
+
+    __test__ = False
+    idp_name: str
+    secret_name: str
+    username: str
+    password: str
+    user_context: str
+    original_context: str
+
+
+def create_test_idp(idp_name: str = "test-htpasswd-idp", secret_name: str = "test-htpasswd-secret") -> UserTestSession:
+    """
+    Creates a new HTPasswd IDP in OpenShift and adds a test user.
+
+    Args:
+        idp_name: Name for the IDP
+        secret_name: Name for the secret
+
+    Returns:
+        UserTestSession object
+    """
+    user_context = ""
+    try:
+        # Save the current context (cluster admin)
+        _, original_context, _ = run_command(command=["oc", "config", "current-context"], check=True)
+        original_context = original_context.strip()
+        LOGGER.info(f"Original context (cluster admin): {original_context}")
+
+        # Generate a unique suffix for this test run
+        unique_suffix = "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
```

Comment:
Luca already created an util for this. generate_random_name(). Can that be used?